### PR TITLE
Create openai client for making requests.

### PR DESCRIPTION
**Pull request by ahamed**

### CODE REVIEW

File: classes/Utils.php [hunk 1]
```diff
@@ -1120,6 +1120,25 @@ public function check_nonce() {
 		}
 	}
 
+	/**
+	 * Check if the openai response has any error or not.
+	 * If there any error then send the error response, otherwise continue.
+	 *
+	 * @param array $response The openai response.
+	 * @return object
```
Comment (shewa12) [inline classes/Utils.php:1128]: I think the return type would be mixed: object|null. `send_response` will just echo json data and die, in that case null will be returned.

File: composer.lock [hunk 2]
```diff
@@ -8,26 +8,26 @@
...
                 "type": "zip",
-                "url": "https://api.github.com/repos/CarbonPHP/carbon-doctrine-types/zipball/99f76ffa36cce3b70a4a6abce41dba15ca2e84cb",
-                "reference": "99f76ffa36cce3b70a4a6abce41dba15ca2e84cb",
+                "url": "https://api.github.com/repos/CarbonPHP/carbon-doctrine-types/zipball/18ba5ddfec8976260ead6e866180bd5d2f71aa1d",
+                "reference": "18ba5ddfec8976260ead6e866180bd5d2f71aa1d",
                 "shasum": ""
             },
             "require": {
-                "php": "^7.4 || ^8.0"
+                "php": "^8.1"
```
Comment (shewa12) [inline composer.lock:24]: Need to revert the changes from composer.lock file

File: classes/Utils.php [hunk 3]
```diff
@@ -1120,6 +1120,26 @@ public function check_nonce() {
...
+	/**
+	 * Check if the openai response has any error or not.
+	 * If there any error then send the error response, otherwise continue.
+	 *
+	 * @since   3.0.0
+	 *
+	 * @param array $response The openai response.
+	 * @return mixed
+	 */
+	public function check_openai_response( array $response ) {
```
Comment (harunollyo) [inline classes/Utils.php:1132]: Why this helper here? It's related to `tutorai` we can move it to `tutorai/Helper.php`

**Review by harunollyo — CHANGES_REQUESTED** [review]

File: classes/Utils.php (hunk 3 above)
Comment (ahamed) [reply classes/Utils.php:1132]: I'll move it later. For now keep this as it is.

**Review by harunollyo — APPROVED** [review]
